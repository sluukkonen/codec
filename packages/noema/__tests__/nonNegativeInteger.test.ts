import { nonNegativeInteger } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

test("should accept non-negative integers", () => {
  expectParseSuccess(nonNegativeInteger, 0)
  expectParseSuccess(nonNegativeInteger, 1)
  expectParseSuccess(nonNegativeInteger, 2)
  expectParseSuccess(nonNegativeInteger, 3)
})

test("should not accept negative integers", () => {
  expectParseFailure(nonNegativeInteger, -1)
  expectParseFailure(nonNegativeInteger, -2)
  expectParseFailure(nonNegativeInteger, -2)
})

test("should not accept non-integers", () => {
  expectParseFailure(nonNegativeInteger, -0.5)
  expectParseFailure(nonNegativeInteger, 0.5)
})

test("should not accept NaN", () => {
  expectParseFailure(nonNegativeInteger, NaN)
})
