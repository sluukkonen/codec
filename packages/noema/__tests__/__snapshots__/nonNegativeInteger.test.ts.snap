// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nonNegativeInteger should not accept NaN 1`] = `
[
  {
    "code": "is_nan",
    "path": [],
  },
]
`;

exports[`nonNegativeInteger should not accept negative integers 1`] = `
[
  {
    "code": "too_small",
    "min": 0,
    "path": [],
  },
]
`;

exports[`nonNegativeInteger should not accept negative integers 2`] = `
[
  {
    "code": "too_small",
    "min": 0,
    "path": [],
  },
]
`;

exports[`nonNegativeInteger should not accept negative integers 3`] = `
[
  {
    "code": "too_small",
    "min": 0,
    "path": [],
  },
]
`;

exports[`nonNegativeInteger should not accept non-integers 1`] = `
[
  {
    "code": "invalid_integer",
    "path": [],
  },
]
`;

exports[`nonNegativeInteger should not accept non-integers 2`] = `
[
  {
    "code": "invalid_integer",
    "path": [],
  },
]
`;
